File: web/Angular4/src/app/Modules/TicketRead/TicketRead.Entity.ts
import {TicketEntity} from "../Ticket/Ticket.Entity";
import {EmployeeEntity} from "../Employee/Employee.Entity";

export class TicketReadEntity {

    ticketId: number;
    employeeId: number;
    status : number;
    ticketEntity : TicketEntity;
    employeeEntity: EmployeeEntity;


    IsEdit: boolean;
    IsActive: boolean = false;
    IsSelected: boolean = false;
    constructor(data: any = null) {
        if (data == null) {
            this.ticketId = null;
            this.employeeId = null;
            this.status = null;
            this.ticketEntity = new TicketEntity();
            this.employeeEntity = new EmployeeEntity();
        } else {
            this.ticketId = data.ticketId;
            this.employeeId = data.employeeId;
            this.status = data.status;
            this.ticketEntity = data.ticketEntity;
            this.employeeEntity = data.employeeEntity;
            if (this.ticketEntity == null) this.ticketEntity = new TicketEntity();
            if (this.employeeEntity == null) this.employeeEntity = new EmployeeEntity();
        }
        this.IsEdit = false;
    }
}

File: src/Module/TicketRead/TicketReadEntity.java
package Module.TicketRead;

import Module.Employee.EmployeeEntity;
import Module.Employee.EmployeeModel;
import Module.Ticket.TicketEntity;
import Module.Ticket.TicketModel;

import java.io.Serializable;

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
public class TicketReadEntity implements Serializable {
    public Integer ticketId;
    public Integer employeeId;
    public Integer status;
    public TicketEntity ticketEntity;
    public EmployeeEntity employeeEntity;


    public TicketReadEntity() {
    }

    public TicketReadEntity(TicketreadModel TicketreadModel, Object... objects) {
        this.ticketId = TicketreadModel.getTicketId();
        this.employeeId = TicketreadModel.getEmployeeId();
        this.status = TicketreadModel.getStatus();
        for (Object object : objects) {
            if (object instanceof TicketModel) {
                this.ticketEntity = new TicketEntity((TicketModel) object);
            }else  if (object instanceof EmployeeModel) {
                this.employeeEntity = new EmployeeEntity((EmployeeModel) object);
            }
        }
    }

    public TicketreadModel toModel() {
        TicketreadModel TicketreadModel = new TicketreadModel();
        TicketreadModel.setEmployeeId(employeeId);
        TicketreadModel.setStatus(status);
        TicketreadModel.setTicketId(ticketId);
        return TicketreadModel;
    }
}
